public/src/embedContext/InteruptionsHandler.js

```diff
@@ -0,0 +1,153 @@
+
+        // scenario 3 - interupted before reaching 80% threshold -
+        /*
+            3A: User returns while the are you still there dialog is active - let them continue only if they continue when the dialog would have been active.
+            3B: User returns while the time out dialog is active but the additional 2 min timer is active - close the task and let them resume from where they left off (if first attempt)
+            3C: User returns after the times out internal timer finishes (1 min for 3 timeouts, 2 mins for are you still there, 2 mins for times up dialog) - Restart the task from the beginning (if first attempt) or submit as complete if not first attempt
+            3D: User is gone for 3-5 mins and there is no active dialog - show the are you still there dialog (or the times up dialog if shown before)
+            3E: User is gone for over 5 mins and theres no active dialog - Show the Times up dialog and restart the task from the beginning (if first attempt) or submit as complete if not.
+            3F: User is gone for less than 3 mins and theres no active dialog - dont let the user interract with the current trial and skip ahead to the next trial.
+        */
+        if (cache.trialNumber >= 2 && cache.trialNumber < completionThresholdTrialNo) {
+            // 3A
+            if (context.bottomScreenPanel && context.bottomScreenPanel.tag == ARE_YOU_THERE_TAG && interuptionLengthMs < twoMinsMs) {
+                // let the user continue from where they are as the dialog would have been still active, no increment to attempt count
+                // the timer will continue to tick down when they return so if they dont continue within the reminaing time they will get the times up dialog.
+                console.log('3A-A');
+                return;
+            } else if (context.bottomScreenPanel && context.bottomScreenPanel.tag == ARE_YOU_THERE_TAG && interuptionLengthMs >= twoMinsMs && interuptionLengthMs < fourMinsMs) {
+                // the user returned after the are you still there dialog would have timed out but before the time out dialogs 2 mins internal countdown has expired,
+                // increment attempt count but they can countinue from where they left off
+                console.log('3A-B');
+                let closeMessage = new CloseMessage(false, true, false);
+                EmbedContext.sendMessage('close', closeMessage.stringify());
+                return;
+            } else if (context.bottomScreenPanel && context.bottomScreenPanel.tag == ARE_YOU_THERE_TAG && interuptionLengthMs >= fourMinsMs) {
+                // the user returned after both the are you still there and times up times up dialogs would have timed out,
+                // increment attempt count and restart from the beginning
+                console.log('3A-C');
+                let closeMessage = new CloseMessage(false, true, true);
+                EmbedContext.sendMessage('close', closeMessage.stringify());
```

Comment:
Similarly here, rather than closing I would replace the 'are you still there' dialog with the 'timed out' dialog instead.